src/audio/google_rtc_audio_processing.c

```diff
@@ -113,11 +127,104 @@ static int google_rtc_audio_processing_reconfigure(struct comp_dev *dev)
 		  config, size);
 
 	cd->reconfigure = false;
-	ret = GoogleRtcAudioProcessingReconfigure(cd->state, config, size);
-	if (ret) {
-		comp_err(dev, "GoogleRtcAudioProcessingReconfigure failed: %d",
-			 ret);
-		return -EINVAL;
+
+	uint8_t *google_rtc_audio_processing_config;
+	size_t google_rtc_audio_processing_config_size;
+	int num_capture_input_channels;
+	int num_capture_output_channels;
+	float aec_reference_delay;
+	float mic_gain;
+	bool google_rtc_audio_processing_config_present;
+	bool num_capture_input_channels_present;
+	bool num_capture_output_channels_present;
+	bool aec_reference_delay_present;
+	bool mic_gain_present;
+
+	GoogleRtcAudioProcessingParseSofConfigMessage(config, size,
+						      &google_rtc_audio_processing_config,
+						      &google_rtc_audio_processing_config_size,
+						      &num_capture_input_channels,
+						      &num_capture_output_channels,
+						      &aec_reference_delay,
+						      &mic_gain,
+						      &google_rtc_audio_processing_config_present,
+						      &num_capture_input_channels_present,
+						      &num_capture_output_channels_present,
+						      &aec_reference_delay_present,
+						      &mic_gain_present);
+
+	if (num_capture_input_channels_present || num_capture_output_channels_present) {
+		if (num_capture_input_channels_present) {
+			cd->num_capture_input_channels = num_capture_input_channels;
+			comp_info(dev,
+				  "google_rtc_audio_processing_reconfigure(): Applying num capture input channels %d",
+				  num_capture_input_channels);
+		}
+		if (num_capture_output_channels_present) {
+			cd->num_capture_output_channels = num_capture_output_channels;
+			comp_info(dev,
+				  "google_rtc_audio_processing_reconfigure(): Applying num capture output channels %d",
+				  num_capture_output_channels);
+		}
+
+		ret = GoogleRtcAudioProcessingSetStreamFormats(cd->state,
+							       GOOGLE_RTC_AUDIO_PROCESSING_SAMPLE_RATE_HZ,
+							       cd->num_capture_input_channels,
+							       cd->num_capture_output_channels,
+							       GOOGLE_RTC_AUDIO_PROCESSING_SAMPLE_RATE_HZ,
+							       cd->num_aec_reference_channels);
+
+		if (ret) {
+			comp_err(dev, "GoogleRtcAudioProcessingSetStreamFormats failed: %d",
+				 ret);
+			return -EINVAL;
+		}
+	}
+
+	if (aec_reference_delay_present || mic_gain_present) {
+		float *capture_headroom_linear_use = NULL;
+		float *echo_path_delay_ms_use = NULL;
+
+		if (mic_gain_present) {
+			capture_headroom_linear_use = &mic_gain;
+
+			comp_info(dev,
+				  "google_rtc_audio_processing_reconfigure(): Applying capture linear headroom: %d.%d",
+				  (int)mic_gain, (int)(100 * mic_gain) - 100 * ((int)mic_gain));
+		}
+		if (aec_reference_delay_present) {
+			echo_path_delay_ms_use = &aec_reference_delay;
+			comp_info(dev,
+				  "google_rtc_audio_processing_reconfigure(): Applying aec reference delay: %d.%d",
+				  (int)aec_reference_delay,
```

Comment:
not sure i grasp what is going on here, you appear to be logging a pointer, is that intended?

Reply:
What I wanted to achieve is to print out a floating point value.
However, it seems like with comp_info, using #f for floating point values is broken, and just printing random values.

This was the best I could do to achieve a log-line with a floating point value

Reply:
Might be worth adding a comment since its not particularly clear

Reply:
Good point.
Done